source/backend/api/Services/CompensationRequisitionService.cs

```diff
@@ -142,7 +151,131 @@ private static string GetCompensationRequisitionStatusText(bool? isDraft)
+            _compensationRequisitionRepository.CommitTransaction();
+
+            return newCompensationRequisition;
+        }
+
+        private List<PimsCompensationRequisition> GetAcquisitionFileCompReqs(long fileId)
+        {
+            _logger.LogInformation("Getting compensations for acquisition file id: {acquisitionFileId}", fileId);
+            _user.ThrowIfNotAuthorized(Permissions.CompensationRequisitionView, Permissions.AcquisitionFileView);
+            _user.ThrowInvalidAccessToAcquisitionFile(_userRepository, _acqFileRepository, fileId);
+
+            return _compensationRequisitionRepository.GetAllByAcquisitionFileId(fileId).ToList();
+        }
+
+        private List<PimsCompensationRequisition> GetLeaseFileCompReqs(long fileId)
+        {
+            _logger.LogInformation("Getting compensations for Lease file id: {LeaseId}", fileId);
+            _user.ThrowIfNotAuthorized(Permissions.LeaseView);
+
+            var pimsUser = _userRepository.GetByKeycloakUserId(_user.GetUserKey());
+            pimsUser.ThrowInvalidAccessToLeaseFile(_leaseRepository.GetNoTracking(fileId).RegionCode);
+
+            return _compensationRequisitionRepository.GetAllByLeaseFileId(fileId).ToList();
+        }
+
+        private PimsCompensationRequisition UpdateAcquisitionFileCompensation(PimsCompensationRequisition compensationRequisition)
+        {
+            var currentCompensation = _compensationRequisitionRepository.GetById(compensationRequisition.CompensationRequisitionId);
+
+            var currentAcquisitionFile = _acqFileRepository.GetById((long)currentCompensation.AcquisitionFileId);
+            var currentAcquisitionStatus = Enum.Parse<AcquisitionStatusTypes>(currentAcquisitionFile.AcquisitionFileStatusTypeCode);
+
+            if (!_acquisitionStatusSolver.CanEditOrDeleteCompensation(currentAcquisitionStatus, currentCompensation.IsDraft) && !_user.HasPermission(Permissions.SystemAdmin))
+            {
+                throw new BusinessRuleViolationException("The file you are editing is not active or hold, so you cannot save changes. Refresh your browser to see file state.");
+            }
+
+            CheckTotalAllowableCompensation(currentAcquisitionFile, compensationRequisition);
+            compensationRequisition.FinalizedDate = CheckFinalizedDate(currentCompensation.IsDraft, compensationRequisition.IsDraft, currentCompensation.FinalizedDate);
+
+            PimsCompensationRequisition updatedEntity = _compensationRequisitionRepository.Update(compensationRequisition);
+            AddAcquisitionNoteIfStatusChanged(compensationRequisition.Internal_Id, (long)compensationRequisition.AcquisitionFileId, currentCompensation.IsDraft, compensationRequisition.IsDraft);
+            _compensationRequisitionRepository.CommitTransaction();
+
+            return updatedEntity;
+        }
+
+        private PimsCompensationRequisition UpdateLeaseFileCompensation(PimsCompensationRequisition compensationRequisition)
```

Comment:
This does seem pretty light on business logic compared to acquisition. Is that by design, or could that be missing from the story?

Reply:
it is by design at this time. I've checked with Dev and we currently don't have more rules.